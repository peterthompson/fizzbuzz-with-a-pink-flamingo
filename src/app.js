import express from "express";
import pinkFlamingo from "./pink-flamingo";

const app = express();

app.get("/", (req, res) => {
  const result = [];

  for (let i = 0; i <= 100; i++) {
    result.push(pinkFlamingo(i));
  }

  res.set("Content-Type", "text/plain");
  res.send(result.join("\n").toString());
});

app.get("/:number", (req, res) => {
  const number = req.params.number;
  const result = pinkFlamingo(number);

  res.set("Content-Type", "text/plain");
  res.send(result.toString());
});

export default app;
