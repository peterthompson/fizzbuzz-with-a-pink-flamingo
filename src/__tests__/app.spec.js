import request from "supertest";
import app from "../app";

describe("app", () => {
  describe("route /", () => {
    it("should return pink flamingo result for numbers 0 to 100", async () => {
      const response = await request(app).get("/");

      expect(response.text).toMatchSnapshot();
    });
  });

  describe("route /:number", () => {
    it("should return pink flamingo result for the numbers", async () => {
      const response = await request(app).get("/0");

      expect(response.text).toEqual("Pink Flamingo");
    });
  });
});
