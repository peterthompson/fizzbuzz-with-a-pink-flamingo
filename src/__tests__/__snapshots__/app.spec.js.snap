// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app route / should return pink flamingo result for numbers 0 to 100 1`] = `
"Pink Flamingo
Flamingo
Flamingo
Flamingo
4
Flamingo
Fizz
7
Flamingo
Fizz
Buzz
11
Fizz
Flamingo
14
FizzBuzz
16
17
Fizz
19
Buzz
Flamingo
22
23
Fizz
Buzz
26
Fizz
28
29
FizzBuzz
31
32
Fizz
Flamingo
Buzz
Fizz
37
38
Fizz
Buzz
41
Fizz
43
44
FizzBuzz
46
47
Fizz
49
Buzz
Fizz
52
53
Fizz
Flamingo
56
Fizz
58
59
FizzBuzz
61
62
Fizz
64
Buzz
Fizz
67
68
Fizz
Buzz
71
Fizz
73
74
FizzBuzz
76
77
Fizz
79
Buzz
Fizz
82
83
Fizz
Buzz
86
Fizz
88
Flamingo
FizzBuzz
91
92
Fizz
94
Buzz
Fizz
97
98
Fizz
Buzz"
`;
